docs/analyzer/adding_recognizers.md

```diff
@@ -0,0 +1,246 @@
+titles_recognizer.analyze(text="Mr. Schmidt",entities="TITLE")
+```
+
+Adding it to the list of recognizers:
+
+```python
+from presidio_analyzer import AnalyzerEngine, RecognizerRegistry
+
+registry = RecognizerRegistry()
+registry.load_predefined_recognizers()
+
+# Add the recognizer to the existing list of recognizers
+registry.add_recognizer(titles_recognizer)
+
+# Set up analyzer with our updated recognizer registry
+analyzer = AnalyzerEngine(registry=registry)
+
+# Run with input text
+text="His name is Mr. Jones"
+results = analyzer.analyze(text=text, language="en")
+print(results)
+
+```
+
+Alternatively, we can add the recognizer directly to the existing registry:
+
+```python
+from presidio_analyzer import AnalyzerEngine
+
+analyzer = AnalyzerEngine()
+
+analyzer.registry.add_recognizer(titles_recognizer)
+
+results = analyzer.analyze(text=text,language="en")
+print(results)
+```
+
+### Creating a new `EntityRecognizer` in code
+
+To create a new recognizer via code:
+
+1. Create a new Python class which implements [LocalRecognizer](../../presidio-analyzer/presidio_analyzer/local_recognizer.py).
+(`LocalRecognizer` implements the base [EntityRecognizer](../../presidio-analyzer/presidio_analyzer/entity_recognizer.py) class)
+
+    This class has the following functions:
+
+    i. load: load a model / resource to be used during recognition
+
+    ```python
+    def load(self)
+    ```
+
+    ii. analyze: The main function to be called for getting entities out of the new recognizer:
+
+    ```python
+    def analyze(self, text, entities, nlp_artifacts)
+    ```
+
+    Notes:
+    1. Each recognizer has access to different NLP assets such as tokens, [lemmas](https://en.wikipedia.org/wiki/Lemma_(morphology)), and more.
+    These are given through the `nlp_artifacts` parameter.
+    Refer to the [source code](../../presidio-analyzer/presidio_analyzer/entity_recognizer.py) for more information.
+
+    2. The `analyze` method should return a list of [RecognizerResult](../../presidio-analyzer/presidio_analyzer/recognizer_result.py).
+
+2. Add it to the recognizer registry using `registry.add_recognizer(my_recognizer)`.
+
+### Creating a remote recognizer
+
+A remote recognizer is an `EntityRecognizer` object interacting with an external service. The external service could be a 3rd party PII detection service or a custom service deployed in parallel to Presidio.
+
+<details>
+  <summary>Click here for a reference implementation of a `RemoteRecognizer`</summary>
+
+Here's an illustrative example of how a `RemoteRecognizer` should be implemented. In this example, an external PII detection service exposes two APIs: `detect` and `supported_entities`. The class implemented here, `MyRemoteRecognizer`, uses the `requests` package to call the external service via HTTP.
+
+In this code snippet, we simulate the external PII detector by using the Presidio analyzer. In reality, we would adapt this code to fit the external PII detector we have in hand.
+
```

Comment:
this can be reaplaced with a link to the class py file, maybe iwth just a one liner on running it (the __main__ )